bacon/model/StorageCouchBaseDB.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  StorageMongoDB.swift
+//  bacon
+//
+//  Underlying class of StorageManager to provide all saving and loading functionalities.
+//  Uses CouchBaseLiteSwfit to provide a full-featured embedded NoSQL database that runs locally.
+//
+//  Created by Travis Ching Jia Yea on 19/3/19.
+//  Copyright © 2019 nus.CS3217. All rights reserved.
+//
+
+import Foundation
+import CouchbaseLiteSwift
+
+private enum DatabaseCollections: String {
+    case transactions
+}
+
+class StorageCouchBaseDB {
+    private var transactionDatabase: Database
+
+    init() throws {
+        // Initialize database
+        transactionDatabase = try StorageCouchBaseDB.openOrCreateEmbeddedDatabase(name: .transactions)
+    }
+
+    private static func openOrCreateEmbeddedDatabase(name: DatabaseCollections) throws -> Database {
+        do {
+            let options = DatabaseConfiguration()
+            // Get the path to the Database
+            let urls = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
+            guard let documentDirectory = urls.first else {
+                throw InitializationError(message: "unable to access document directory!")
+            }
+            let databaseFolderUrl = documentDirectory.appendingPathComponent(name.rawValue, isDirectory: true)
+            let databaseFolderPath = try getOrCreateFolderPath(for: databaseFolderUrl)
+            // Set the folder path for the CouchbaseLite Database
+            options.directory = databaseFolderPath
+            // Create a new database or get handle to existing database at specified path
+            return try Database(name: name.rawValue, config: options)
+        } catch let error {
```

Comment:
You don't have to specify `let error`, swift automatically binds the error object to a local `error` variable: https://docs.swift.org/swift-book/LanguageGuide/ErrorHandling.html

Reply:
Oh cool!